utils/shonenjump/Makefile

```diff
@@ -0,0 +1,47 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=shonenjump
+PKG_VERSION:=0.8.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/suzaku/$(PKG_NAME)/tar.gz/v$(PKG_VERSION)?
+PKG_HASH:=ff02e0872c96dbaecdb64e8ad977a98260ec1a105869d8cf4b988ac579dcf789
+
+PKG_LICENSE:=BSD-3-Clause
+PKG_LICENSE_FILES:=LICENSE
+PKG_MAINTAINER:=John Audia <therealgraysky@proton.me>
+
+PKG_BUILD_DEPENDS:=golang/host
+PKG_BUILD_PARALLEL:=1
+
+GO_PKG:=github.com/suzaku/shonenjump
+GO_PKG_BUILD_PKG:=$(GO_PKG)
+
+include $(INCLUDE_DIR)/package.mk
+include ../../lang/golang/golang-package.mk
+
+define Package/shonenjump
+  SECTION:=utils
+  CATEGORY:=Utilities
+  TITLE:=Shonenjump
+  URL:=https://github.com/suzaku/shonenjump
+  DEPENDS:=$(GO_ARCH_DEPENDS) +libstdcpp
+endef
+
+define Package/shonenjump/description
+ A lightweight autojump clone written in Go.
+endef
+
+define Package/shonenjump/install
+	$(call GoPackage/Package/Install/Bin,$(1))
+	$(INSTALL_DIR) $(1)/etc/profile.d
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/scripts/shonenjump.bash $(1)/etc/profile.d
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/scripts/shonenjump.fish $(1)/etc/profile.d
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/scripts/shonenjump.zsh $(1)/etc/profile.d
```

Comment:
I don't think you can just copy these scripts into /etc/profile.d and be done. Most if not all shells will load these scripts - at best these pollute the environment, at worst they cause errors due to incompatible syntax (e.g. busybox sh loading the fish/zsh scripts)

I think as a minimum you need to add stub script(s) that load the right shonenjump script for the user's shell. Better would be to split the scripts into separate packages, so if I use bash I just need to install the bash package and not have to install the other scripts.

Reply:
I like these suggestions (split package and have a stub script automatically source these on a per-shell basis).  I have a few questions:
1. If not `/etc/profile.d/` where should these script go?
2. Can you suggest a stub script that would load these automatically?  I am not familiar enough with how OpenWrt would call these.

Reply:
> If not /etc/profile.d/ where should these script go? (/usr/share/shonenjump/)?

/usr/share/shonenjump would be fine.

> Can you suggest a stub script that would load these automatically?

Test for a specific shell, if it is found then source the corresponding shonenjump script - something like [this](https://github.com/openwrt/openwrt/blob/afa229038c05ba0ca20595d7f73bea94db21d3a6/package/base-files/files/etc/shinit#L6).

Ensure the stub script(s) only use POSIX shell features so they can be safely loaded with any POSIX-compatible shell.

> How can we make selecting the base file automatically select the one for bash but allow the user to deselect that one if another is selected?

You don't - users should install the shell-specific package (which pulls in the base package), not the base package directly. The base package description can include instructions for users to do this.

Reply:
> Test for a specific shell, if it is found then source the corresponding shonenjump script - something like [this](https://github.com/openwrt/openwrt/blob/afa229038c05ba0ca20595d7f73bea94db21d3a6/package/base-files/files/etc/shinit#L6).

I will look at the example you provided.

> users should install the shell-specific package (which pulls in the base package), not the base package directly. The base package description can include instructions for users to do this.

Can you suggest the proper syntax for a base package and for the three shell-specific packages?  I am thinking this split might be more trouble than it's worth; savings is ~5 K:
```
-rw-r--r-- 1 root root 2.9K Jan 14  2023 shonenjump.bash
-rw-r--r-- 1 root root 2.5K Jan 14  2023 shonenjump.fish
-rw-r--r-- 1 root root 2.1K Jan 14  2023 shonenjump.zsh
```
What do you think?

Reply:
> Where would such a file go such that upon login it is called like /etc/shinit gets called?
> What if the user is not using that particular shell? Just searching for the executable seems like a bad idea to check.
> Can you suggest the proper syntax for a base package and for the three shell-specific packages?

You should do some work and find the answers for yourself.

> I am thinking this split might be more trouble than it's worth; savings is ~5 K:

Having shell-specific packages is optional, loading the right script for the right shell is the key point.

Reply:
> You should do some work and find the answers for yourself.

My thought was to use `/etc.profile.d/` but that will cause the default shell to source the needed files whereas we need each of the respective shells (bash, fish, or zsh) to source.  I am not sure how to do this without studying each of the respective shell's packages.

Reply:
Actually I was thinking about this today

> Test for a specific shell, if it is found then source the corresponding shonenjump script - something like [this](https://github.com/openwrt/openwrt/blob/afa229038c05ba0ca20595d7f73bea94db21d3a6/package/base-files/files/etc/shinit#L6).
> 
> Ensure the stub script(s) only use POSIX shell features so they can be safely loaded with any POSIX-compatible shell.

This is wrong.

> that will cause the default shell to source the needed files whereas we need each of the respective shells (bash, fish, or zsh) to source.

You are correct.

> I am not sure how to do this without studying each of the respective shell's packages.

Not sure how much you will get from studying each shell's packages, but you will at least need to read the man pages for each shell to find out which files they load on startup (for all interactive shells, not just login shells).

Reply:
In reviewing the man pages for zsh, it seems that placing the source line in /etc/zsh/zshrc would get the job done. It works in this fashion on Arch Linux but our zsh package seems to ignore that file entirely. I have not looked into fish or bash. Since upstream's [docs](https://github.com/suzaku/shonenjump#other-distros) calls out a manual modification to ~/.shellprofile, I think shipping this without trying to modify things is the right thing to do.

Reply:
Whether you want to do the work is up to you.

If I am installing software from source, then I would expect that I might have to do some integration work, e.g. write an init.d script/systemd unit file, edit my .bashrc file, etc. If I am installing software from a distro package, then I would expect most of that integration work to already be done.

Reply:
I understand your expectation.  If OpenWrt has this philosophy for packaging, it makes sense to continue digging for the replace place for a script to place a file.  I am out of ideas that don't involve directly editing another package's file.